### PR TITLE
156 Resend email verification

### PR DESCRIPTION
**Pull request by victoriachan**
Ticket: https://national-archives.atlassian.net/jira/software/projects/SSO/boards/92?selectedIssue=SSO-156

### CODE REVIEW

File: tna_account_management/users/forms.py [hunk 1]
```diff
@@ -5,6 +5,21 @@
...
+class VerifyEmailForm(HelperMixin, forms.Form):
+    @property
+    def helper(self):
+        fh = super().helper
+        fh.layout.extend([
+            Button.primary(
+                name="submit",
+                type="submit",
+                # TODO: Should be: "Resend email to <user.email>"
+                value="Resend email",
```
Comment (ababic) [inline tna_account_management/users/forms.py:17]: Can the view pass the button label to the form somehow?

File: tna_account_management/users/models.py [hunk 2]
```diff
@@ -177,6 +177,9 @@ def check_password(self, raw_password: str) -> bool:
         except Exception:
             False
 
+    def resend_email_verification(self):
```
Comment (ababic) [inline tna_account_management/users/models.py:180]: Nitpick: 
```suggestion
    def resend_verification_email(self):
```

File: tna_account_management/project_styleguide/templates/patterns/pages/user/verify_email.html [hunk 3]
```diff
@@ -1,2 +1,11 @@
...
-{% block title %}Verify your email address{% endblock %}
+{% load crispy_forms_tags %}
+
+{% block content %}
+    <p>You would have received an email from The National Archives with instructions on how to verify your email address.</p>
+    <p>Please check your inbox and follow the instructions to verify your account.</p>
+
+    {% if form %}
+        {% crispy form %}
+    {% endif %}
```
Comment (ababic) [inline tna_account_management/project_styleguide/templates/patterns/pages/user/verify_email.html:10]:
```suggestion
    {{ block.super }}
```

File: tna_account_management/project_styleguide/templates/patterns/pages/user/verify_email.html [hunk 4]
```diff
@@ -1,2 +1,11 @@
 {% extends "patterns/base_form_view.html" %}
-{% block title %}Verify your email address{% endblock %}
+{% load crispy_forms_tags %}
```
Comment (ababic) [inline tna_account_management/project_styleguide/templates/patterns/pages/user/verify_email.html:2]: When using `{{ block.super }}`, we don't need to re-import this

File: tna_account_management/users/views.py [hunk 5]
```diff
@@ -47,10 +47,27 @@ class AccountDashboardView(LoginRequiredMixin, CommonContextMixin, TemplateView)
...
+    form_class = forms.VerifyEmailForm
     template_name = "patterns/pages/user/verify_email.html"
     success_url = reverse_lazy("dashboard")
+    
+    def form_valid(self, form):
+        user = self.request.user
+        try:
+           user.resend_email_verification()
+        except Exception:
+            logger.exception(f"Failed to send verification email to {user.email}")
```
Comment (ababic) [inline tna_account_management/users/views.py:61]: We should avoid including emails in the message, as it is sensitive data. Sentry can be configured to remove PII, but not if it's in the error message itself.

**Review by ababic — CHANGES_REQUESTED** [review]
This is much better, thanks @victoriachan. Just a couple of things to consider.

File: tna_account_management/users/forms.py (hunk 1 above)
Comment (victoriachan) [reply tna_account_management/users/forms.py]: I've updated this now to use  an arg passed in from form init. Not sure if this is the best approach. What do you think?

File: tna_account_management/users/views.py (hunk 5 above)
Comment (victoriachan) [reply tna_account_management/users/views.py]: Ah ok. Thanks. I've removed it now.

**Comment by victoriachan** [comment]
Hi @ababic I've made some changes, can please review the bit about passing in the email to the form button? Thanks.

**Review by ababic — CHANGES_REQUESTED** [review]
Spot on! thanks @victoriachan!

File: tna_account_management/users/forms.py (hunk 1 above)
Comment (ababic) [reply tna_account_management/users/forms.py]: Peeerfect :)

**Pull request merged**